waku/node/waku_node.nim

```diff
@@ -361,22 +365,50 @@ proc filterHandleMessage*(node: WakuNode, pubsubTopic: PubsubTopic, message: Wak
   await node.wakuFilter.handleMessage(pubsubTopic, message)
   await node.wakuFilterLegacy.handleMessage(pubsubTopic, message) #TODO: remove legacy
 
-
-proc mountFilterClient*(node: WakuNode) {.async, raises: [Defect, LPError].} =
+proc mountFilterClient*(node: WakuNode, filterV2MessagePushHandler= none(filter_client.MessagePushHandler)) {.async, raises: [Defect, LPError].} =
+  ## Mounting both filter clients v1 - legacy and v2.
+  ## Giving option for application level to chose btw own push message handling or
+  ## rely on node provided cache. - This only applies for v2 filter client
   info "mounting filter client"
 
   node.wakuFilterClientLegacy = WakuFilterClientLegacy.new(node.peerManager, node.rng)
+
+  if filterV2MessagePushHandler.isNone():
+    # TODO: tie it to config, but from proper place (rest_legacy_filter_api.filterMessageCacheDefaultCapacity)
+    node.filterV2PushMessageCache = TopicCache.init(capacity=30)
+
+    let internalPushMessageHandler: filter_client.MessagePushHandler = proc(pubsubTopic: string, message: WakuMessage) {.gcsafe, closure.} =
+      node.filterV2PushMessageCache.addMessage(pubsubTopic, message)
+
+    node.wakuFilterClient = filter_client.WakuFilterClient.new(node.rng, internalPushMessageHandler, node.peerManager)
+  else:
+    node.wakuFilterClient = filter_client.WakuFilterClient.new(node.rng, filterV2MessagePushHandler.get(), node.peerManager)
```

Comment:
Not sure I see the need here for this internal vs external message cache and handlers? For one, I think we're overcomplicating the responsibility of the `WakuNode` object, which should not really contain more logic or properties than what is necessary to mount the different protocols on the `switch` it encapsulates. In other words, it makes sense to me to have the application layer always provide its own message handler when mounting a filter client. It's up to this application layer if it needs to cache messages or not. In this case, the application layer is the REST API, which needs a handler and a cache to allow for polling functionality. I'd rather simplify this by removing the cache from the node and the internal handler unless there is some need/reasoning behind it that requires it?